pkg/cmd/server/middleware.go

```diff
@@ -172,9 +176,9 @@ func (mc *MiddlewareChain[T]) validate(mod MiddlewareModification[T]) error {
 	}
 
 	// prevent appending/prepending a duplicate middleware
-	for _, middleware := range mod.Middlewares {
-		if existingNames.Has(middleware.Name) && mod.DependencyMiddlewareName == middleware.Name && mod.Operation != OperationReplace {
-			return fmt.Errorf("modification will cause a duplicate in chain: %s", middleware.Name)
+	for _, mw := range mod.Middlewares {
+		if existingNames.Has(mw.Name) && mod.DependencyMiddlewareName == mw.Name && mod.Operation != OperationReplace {
+			return fmt.Errorf("modification will cause a duplicate in chain: %s", mw.Name)
```

Comment:
make a spiceerrors.MustBugf

Reply:
That does not work - this is only executed only in tests, and if I have a panic here, because it happened in a random goroutine in the gRPC server, I couldn't recover that panic in the test I wrote to ensure these new helpers all work.

Reply:
Ok